phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1159,6 +1153,29 @@ boolean ensureNamespaceCreated(String schemaName) throws SQLException {
         return createdNamespace;
     }
 
+    /**
+     * Returns true if HBase namespace exists, else returns false
+     * @param hbaseAdmin HbaseAdmin Object
+     * @param schemaName Phoenix schema name for which we check existence of the HBase namespace
+     * @return true if the HBase namespace exists, else returns false
+     * @throws SQLException If there is an exception checking the HBase namespace
+     */
+    boolean isHbaseNamespaceAvailable(HBaseAdmin hbaseAdmin, String schemaName) throws IOException{
+        boolean namespaceExists = false;
```

Comment:
Need a test for this logic.

Reply:
Added unit tests in ServerUtilTest class.